### PR TITLE
Do not call the config endpoint if API is lower than 1.30

### LINKED ISSUE #180: "docker stack rm" results in "page not found"
I'm guessing this is related to https://github.com/docker/cli/pull/162

I'm using a master build which I think has the fix for #162 

```
C:\Users\friism>docker version
Client:
 Version:      17.06.0-dev
 API version:  1.29 (downgraded from 1.30)
 Go version:   go1.8.3
 Git commit:   f82f61e
 Built:        Sun Jun 11 23:59:54 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

```
C:\code\atsea-sample-shop-app>docker service ls
ID                  NAME                       MODE                REPLICAS            IMAGE                             PORTS
9sr4ophfuwmm        atsea_reverse_proxy        replicated          0/1                 friism/reverse_proxy:latest
hwcgsqn3kio3        atsea_database             replicated          1/1                 friism/atsea_db:latest            *:5432->5432/tcp
ov06vkl99x6c        dockercloud-server-proxy   global              1/1                 dockercloud/server-proxy:latest   *:2376->2376/tcp
s45iwlnfbskp        atsea_payment_gateway      replicated          1/1                 friism/payment_gateway:latest
w8v9yf4aunyv        atsea_appserver            replicated          1/1                 friism/atsea_app:latest           *:5005->5005/tcp,*:8080->8080/tcp

C:\code\atsea-sample-shop-app>docker stack ls
NAME                SERVICES
atsea               4

C:\code\atsea-sample-shop-app>docker stack rm atsea
Error response from daemon: page not found

version: "3.1"

services:
  reverse_proxy:
    build: ./reverse_proxy
    image: friism/reverse_proxy

  database:
    build:
       context: ./database
    image: friism/atsea_db
    environment:
      POSTGRES_USER: gordonuser
      POSTGRES_DB_PASSWORD_FILE: /run/secrets/postgres_password
      POSTGRES_DB: atsea
    ports:
      - "5432:5432"
    networks:
      - back-tier
    secrets:
      - postgres_password

  appserver:
    build:
       context: app
       dockerfile: Dockerfile
    image: friism/atsea_app
    ports:
      - "8080:8080"
      - "5005:5005"
    networks:
      - front-tier
      - back-tier
    secrets:
      - postgres_password

  payment_gateway:
    build:
      context: payment_gateway
    image: friism/payment_gateway
    networks:
      - payment
    secrets:
      - payment_token

networks:
  front-tier:
  back-tier:
  payment:
    driver: overlay

secrets:
  postgres_password:
    file: ./devsecrets/postgres_password
  payment_token:
    file: ./devsecrets/payment_token

C:\code\atsea-sample-shop-app>
```

### PR DESCRIPTION
**Pull request by vdemeester**
Another would be to handle `Page not found` error in those cases 👼
(and https://github.com/moby/moby/pull/33571 could help too 👼)

Fix #180 

/cc @mavenugo @friism 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: cli/command/stack/common.go [hunk 1]
```diff
@@ -67,6 +68,13 @@ func getStackConfigs(
 	apiclient client.APIClient,
 	namespace string,
 ) ([]swarm.Config, error) {
+	version, err := apiclient.ServerVersion(ctx)
+	if err != nil {
+		return []swarm.Config{}, err
+	}
+	if versions.LessThan(version.APIVersion, "1.30") {
+		return []swarm.Config{}, nil
+	}
```
Comment (thaJeztah) [inline cli/command/stack/common.go]: I wonder if we should still print a warning (otherwise people may not be aware an option was ignored)

**Comment by thaJeztah** [comment]
Perhaps make the warning more consistent with the error messages we have (and mention which API version is needed?) https://github.com/moby/moby/pull/33571/files#diff-c91a3260cd755746371aa100f9b599a6R24

Reading on my phone, haven checked if the docker-compose / stack deploy messages are different 😊

File: cli/command/stack/remove.go [hunk 2]
```diff
@@ -55,10 +56,20 @@ func runRemove(dockerCli command.Cli, opts removeOptions) error {
...
 
-		configs, err := getStackConfigs(ctx, client, namespace)
+		var configs []swarm.Config
+
+		version, err := client.ServerVersion(ctx)
 		if err != nil {
 			return err
 		}
+		if versions.LessThan(version.APIVersion, "1.30") {
+			fmt.Fprintf(dockerCli.Err(), "WARNING: requires API version 1.30, but the Docker daemon API version is %s", version.APIVersion)
```
Comment (dnephin) [inline cli/command/stack/remove.go]: What requires API version 1.30? If i saw this message from `docker stack rm` I would be confused. I think we need to include that this is about "configs" only.

Comment (thaJeztah) [inline cli/command/stack/remove.go]: Oh, I was discussing options with @vdemeester this morning; guess I should've put them here, as he didn't come to updating yet 👍

**Comment by vdemeester** [comment]
@dnephin @thaJeztah updated 👼

**Comment by thaJeztah** [comment]
oh! forgot to actually LGTM.. well, LGTM! thanks :joy:

**Pull request merged**